src/ucp/rndv/rndv_put.c

```diff
@@ -583,17 +583,23 @@ ucp_proto_rndv_put_mtype_probe(const ucp_proto_init_params_t *init_params)
     size_t frag_size;
     unsigned flags;
     ucp_md_index_t UCS_V_UNUSED dummy_md_id;
+    ucs_memory_type_t frag_mem_type;
     ucp_memory_info_t frag_mem_info;
 
     if (init_params->rkey_config_key == NULL) {
         return;
     }
 
+    frag_mem_type = init_params->rkey_config_key->mem_type;
+    if (!UCS_BIT_GET(context->config.ext.rndv_frag_mem_types, frag_mem_type)) {
+        return;
+    }
```

Comment:
i assume we get RTR with managed memory type because we support ODP for managed memory on GH now? thus can pack rkey for managed memory?
Maybe the rndv/put/mtype should iterate over all possible rndv frag mem types, as the rtr/mtype protocol does?

Reply:
i'd suggest to implement simpler (and lower risk) fix and backport it to 1.18 (discussed with Raul offline). Then we could try to iterate over all possible types.
Potential problems with iterating over all mem types:
1) There are multiple issues if both peers doing ppln using different fragment types (mainly because we allocate different size fragments for different types)
2) There will be noticably more rndv variants, which would make perf charts unreadable
3) Once first 2 problems are resolved we would need to do full QA cycle to verify, which does not align with the 1.18 time frame